assembly/models.ts

```diff
@@ -2,17 +2,63 @@ import { context, PersistentMap, PersistentSet } from "near-sdk-as";
 
 type AccountId = string;
 
+const NFT_SPEC = "nft-1.0.0";
+const NFT_NAME = "Share";
+const NFT_SYMBOL = "SHARE";
+
+@nearBindgen
+export class NFTContractMetadata {
+    spec: string = NFT_SPEC;
+    name: string = NFT_NAME;
+    symbol: string = NFT_SYMBOL;
+    icon: string;
+    base_uri: string;
+    reference: string;
+    reference_hash: string;
+}
+
+@nearBindgen
+class TokenMetadata {
+    constructor(
+        public title: string = "",
+        public issued_at: string = "",
+        public copies: number = 1,
+        public extra: string = "", 
+        public description: string = "",
+        public media: string = "",
+        public media_hash: string = "",
+        public expires_at: string = "",
+        public starts_at: string = "",
+        public updated_at: string = "",
+        public reference: string = "",
+        public reference_hash: string = "",
+    ) {}
+  }
+
 @nearBindgen
 export class Design {
-    owner: string;
+    id: string;
+    owner_id: string;
+    metadata: TokenMetadata;
     constructor(
-        public instructions: Array<i32>,
+        instructions: Array<i32>,
         public seed: i32, 
     ) {
-        this.owner = context.sender;
+        this.id = context.sender;
+        this.owner_id = context.sender;
+
+       
+        const title = `${seed}`; 
+        const issued_at = context.blockTimestamp.toString();
+        const copies : number = 1;
+        const extra = `"{'instructions': '${instructions}'}"`;
```

Comment:
I think template strings don't work right now unfortunately

Note that you can define a class with `nearBindgen` and just have it serialised into Uint8Array with JSON.

Reply:
Thanks,
been thinking about this but didn't know how to approach it.

To sum it up (and see if I got it); So now I get an `Extra` `Object` that I then `encode()` into `Uint8Array` and pass to the new `Token` and it ends up being saved in memory as `base64` string that can be then converted to `JSON` on front-end as needed?